flink-connector-mysql-cdc/src/main/java/com/ververica/cdc/connectors/mysql/source/assigners/MySqlSnapshotSplitAssigner.java

```diff
@@ -144,9 +144,30 @@ private MySqlSnapshotSplitAssigner(
     @Override
     public void open() {
         chunkSplitter = createChunkSplitter(sourceConfig, isTableIdCaseSensitive);
+        discoveryCaptureTables();
```

Comment:
The changes in this PR #1449 missed.

Reply:
After rethink I found the optimization only happen when upgrade from 2.0 to 2.3, the benefit is too small and it also let the code too difficult to understand. Thus I decide to ignore it.